PROTO_tests/tests/operator_test.py

```diff
@@ -0,0 +1,774 @@
+        assert ak.power(a, 3, ak.zeros(5, bool)).to_list() == a.to_list()
+
+        # Test a singleton with a mixed Boolean argument
+        a = ak.arange(10)
+        assert [i if i % 2 else i**2 for i in range(10)] == ak.power(a, 2, a % 2 == 0).to_list()
+
+        # Test invalid input, negative
+        n = np.array([-1.0, -3.0])
+        a = ak.array(n)
+        assert np.allclose(ak.power(a, 0.5).to_ndarray(), np.power(n, 0.5), equal_nan=True)
+
+        # Test edge case input, inf
+        infs = [np.inf, -np.inf]
+        assert (np.power(np.array(infs), 2) == ak.power(ak.array(infs), 2).to_ndarray()).all()
+
+    def test_pda_sqrt(self):
+        n = np.array([4, 16.0, -1, 0, np.inf])
+        a = ak.array(n)
+        assert np.allclose(ak.sqrt(a).to_ndarray(), np.sqrt(n), equal_nan=True)
+
+        # Test with a mixed Boolean array
+        a = ak.arange(5)
+        assert [i if i % 2 else i**0.5 for i in range(5)] == ak.sqrt(a, a % 2 == 0).to_list()
+
+    def test_uint_and_bigint_operation_equals(self):
+        u_arr = ak.array(
+            [0, 1, 2, 3, 4, 2**64 - 5, 2**64 - 4, 2**64 - 3, 2**64 - 2, 2**64 - 1],
+            dtype=ak.uint64,
+        )
+        bi_arr = ak.array(
+            [0, 1, 2, 3, 4, 2**64 - 5, 2**64 - 4, 2**64 - 3, 2**64 - 2, 2**64 - 1],
+            dtype=ak.bigint,
+            max_bits=64,
+        )
+        np_arr = np.array(
+            [0, 1, 2, 3, 4, 2**64 - 5, 2**64 - 4, 2**64 - 3, 2**64 - 2, 2**64 - 1],
+            dtype=np.uint,
+        )
+        i_arr = ak.arange(10)
+        f_arr = ak.linspace(1, 5, 10)
+        b_arr = i_arr % 2 == 0
+        u = np.uint(7)
+        i = 7
+        f = 3.14
+        b = True
+
+        # test uint opequals uint functionality against numpy
+        for arr in u_arr, bi_arr, np_arr:
+            arr += u
+        assert u_arr.to_list() == np_arr.tolist()
+        assert bi_arr.to_list() == np_arr.tolist()
+
+        for arr in u_arr, bi_arr, np_arr:
+            arr += arr
+        assert u_arr.to_list() == np_arr.tolist()
+        assert bi_arr.to_list() == np_arr.tolist()
+
+        for arr in u_arr, bi_arr, np_arr:
+            arr -= u
+        assert u_arr.to_list() == np_arr.tolist()
+        assert bi_arr.to_list() == np_arr.tolist()
+
+        for arr in u_arr, bi_arr, np_arr:
+            arr -= arr
+        assert u_arr.to_list() == np_arr.tolist()
+        assert bi_arr.to_list() == np_arr.tolist()
+
+        # redeclare minus self zeroed out
+        u_arr = ak.array(
+            [0, 1, 2, 3, 4, 2**64 - 5, 2**64 - 4, 2**64 - 3, 2**64 - 2, 2**64 - 1],
+            dtype=ak.uint64,
+        )
+        bi_arr = ak.array(
+            [0, 1, 2, 3, 4, 2**64 - 5, 2**64 - 4, 2**64 - 3, 2**64 - 2, 2**64 - 1],
+            dtype=ak.bigint,
+            max_bits=64,
+        )
+        np_arr = np.array(
+            [0, 1, 2, 3, 4, 2**64 - 5, 2**64 - 4, 2**64 - 3, 2**64 - 2, 2**64 - 1],
+            dtype=np.uint,
+        )
```

Comment:
Would it make sense to add a helper function?

Reply:
sure! I can make that change